File: RiceCooker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.RiceCooker = void 0;
var RiceCooker = /** @class */ (function () {
    function RiceCooker(pluggedIn, status, maxCapacity, quantity, cookTime, waterAmount) {
        if (pluggedIn === void 0) { pluggedIn = false; }
        if (status === void 0) { status = 'Power Off'; }
        if (maxCapacity === void 0) { maxCapacity = 3000; }
        if (quantity === void 0) { quantity = 0; }
        if (cookTime === void 0) { cookTime = 0; }
        if (waterAmount === void 0) { waterAmount = 0; }
        this.pluggedIn = pluggedIn;
        this.status = status;
        this.maxCapacity = maxCapacity;
        this.quantity = quantity;
        this.cookTime = cookTime;
        this.waterAmount = waterAmount;
    }
    RiceCooker.prototype.plug = function () {
        this.pluggedIn = !this.pluggedIn;
        console.log("Plugged in: ".concat(this.pluggedIn ? 'On' : 'Off'));
    };
    RiceCooker.prototype.setQuantity = function (newQuantity) {
        if (newQuantity > this.maxCapacity) {
            console.log("Error: quantity cannot exceed maximum capacity of ".concat(this.maxCapacity));
        }
        else {
            this.quantity = newQuantity;
            console.log("Quantity is now: ".concat(this.quantity));
        }
    };
    RiceCooker.prototype.setCookTime = function (newCookTime) {
        if (newCookTime <= 0) {
            console.log('Error: cook time must be positive');
        }
        else {
            this.cookTime = newCookTime;
            console.log("Cook time is now: ".concat(this.cookTime));
        }
    };
    RiceCooker.prototype.setWaterAmount = function (newWaterAmount) {
        if (newWaterAmount <= 0) {
            console.log('Error: water amount must be positive');
        }
        else {
            this.waterAmount = newWaterAmount;
            console.log("Water amount is now: ".concat(this.waterAmount));
        }
    };
    RiceCooker.prototype.cookRice = function () {
        if (this.quantity <= 0) {
            console.log('You need to put the rice you want to cook in');
            return;
        }
        if (this.waterAmount <= 0) {
            console.log('You need to put water to be able to cook rice');
            return;
        }
        if (!this.pluggedIn) {
            console.log('Please turn on the rice cooker');
            return;
        }
        if (this.quantity > this.maxCapacity) {
            console.log('Too much rice, the cooker cannot handle it');
            return;
        }
        console.log("Cooking ".concat(this.quantity, "g of rice"));
        console.log("This will take approximately ".concat(this.cookTime, " minutes"));
    };
    return RiceCooker;
}());
exports.RiceCooker = RiceCooker;

File: rice_cooker.rb
# frozen_string_literal: true


# Class to represent a Rice Cooker
class RiceCooker
  attr_accessor :plugged_in, :is_used, :max_capacity, :quantity, :cook_time, :water_amount

  def initialize(
    plugged_in: false,
    is_used: false,
    max_capacity: 3000,
    quantity: 0,
    cook_time: 0,
    water_amount: 0
  )
    @plugged_in = plugged_in
    @is_used = is_used
    @max_capacity = max_capacity
    @quantity = quantity
    @cook_time = cook_time
    @water_amount = water_amount
  end

  def plug
    @plugged_in = !@plugged_in
    puts "Plugged in: #{@plugged_in ? 'On' : 'Off'}"
  end

  def add_quantity(new_quantity)
    if new_quantity > @max_capacity
      puts "Error: quantity cannot exceed maximum capacity of #{@max_capacity}"
    else
      @quantity = new_quantity
      puts "Quantity is now: #{@quantity}"
    end
  end

  def define_cook_time(new_cook_time)
    if new_cook_time <= 0
      puts 'Error: cook time must be positive'
    else
      @cook_time = new_cook_time
      puts "Cook time is now: #{@cook_time}"
    end
  end

  def define_water_amount(new_water_amount)
    if new_water_amount <= 0
      puts 'Error: water amount must be positive'
    else
      @water_amount = new_water_amount
      puts "Water amount is now: #{@water_amount}"
    end
  end

  def cook_rice
    if @is_used
      puts 'Already in used'
      return
    end

    if @quantity <= 0
      puts 'You need to put the rice you want to cook in'
      return
    end

    if @water_amount <= 0
      puts 'You need to put water to be able to cook rice'
      return
    end

    unless @plugged_in
      puts 'Please turn on the rice cooker'
      return
    end

    if @quantity > @max_capacity
      puts 'Too much rice, the cooker cannot handle it'
      return
    end

    puts "Cooking #{@quantity}g of rice"
    puts "This will take approximately #{@cook_time} minutes"

    @is_used = true
    change_status
  end

  def change_status
    @status = @is_used ? 'In Use' : 'Idle'
    puts "Status changed to: #{@status}"
  end
end
